lib/libmbedtls/core/ecc.c

```diff
@@ -379,3 +342,76 @@ TEE_Result crypto_acipher_ecc_shared_secret(struct ecc_keypair *private_key,
 	mbedtls_ecdh_free(&ecdh);
 	return res;
 }
+
+static const struct crypto_ecc_keypair_ops ecc_keypair_ops = {
+	.generate = &ecc_generate_keypair,
+	.sign = &ecc_sign,
+	.shared_secret = &ecc_shared_secret,
+};
+
+TEE_Result crypto_asym_alloc_ecc_keypair(struct ecc_keypair *s,
+					 uint32_t key_type,
+					 size_t key_size_bits)
+{
+	switch (key_type) {
+	case TEE_TYPE_SM2_DSA_KEYPAIR:
+	case TEE_TYPE_SM2_PKE_KEYPAIR:
+	case TEE_TYPE_SM2_KEP_KEYPAIR:
+		return TEE_ERROR_NOT_IMPLEMENTED;
+	default:
+		break;
+	}
+
+	memset(s, 0, sizeof(*s));
+	s->d = crypto_bignum_allocate(key_size_bits);
+	if (!s->d)
+		goto err;
+	s->x = crypto_bignum_allocate(key_size_bits);
+	if (!s->x)
+		goto err;
+	s->y = crypto_bignum_allocate(key_size_bits);
+	if (!s->y)
+		goto err;
+
+	s->ops = &ecc_keypair_ops;
+
+	return TEE_SUCCESS;
+err:
```

Comment:
empty line above, and another one above `return TEE_ERROR_OUT_OF_MEMORY;` below.

Reply:
ok